project/ws/app/src/lib/routes/app-toc/components/app-toc-home/app-toc-home.component.html

```diff
@@ -66,9 +67,10 @@
                       !(content?.primaryCategory === primaryCategory.PROGRAM) &&
                       !(content?.primaryCategory === primaryCategory.MANDATORY_COURSE_GOAL)">
           <ng-container *ngIf="content?.primaryCategory !== primaryCategory.RESOURCE  && !enrollBtnLoading">
-            <a class="flex action-button justify-center resume" *ngIf="!forPreview || isInIFrame"  (click)="handleAutoBatchAssign()">
-              <ng-container >
-                {{ 'apptochome.enroll' | translate }}
+            <a class="flex action-button justify-center resume" *ngIf="!forPreview || isInIFrame"
+              (click)="openLangDialog($event)">
```

Comment:
change the logic to check if the course is having languages then only show the popup, otherwise call handle auto batch as before